dts/bindings/sensor/st,stm32-qdec.yaml

```diff
@@ -16,6 +16,13 @@ properties:
   pinctrl-names:
     required: true
 
+  st,encoder-mode:
+    type: int
+    description: |
+      Set encoder mode. The value depends on the MCU series. A description of
+      the values can be found in the reference manual or the HAL header. If not
+      set a default value is chosen by the HAL encoder driver.
```

Comment:
Several issues with this description:
Values should be documented. If there is no ambiguity by reading the RM, it should be doable to document them.
If values depends on series, then you might need to update the code to allow a single set of values that are then properly encoded by the driver.
For default value, be more explicit, at least we should get a sense of the functional impact.

See ` st,input-filter-level` as a good example.
See also that, when doable, providing an enum allows early dts level checks.